clarify record creation process (#2820)
astronomer/docs: astro/manage-domains.md

3. In the **Domain** field, enter the domain that you want to map.
4. Click **Create**. The domain is added to your **Managed Domains** and marked as **Unverified**.

> 5. In the entry for your domain, click **Verify**.

Reviewer: ```suggestion
5. In the entry for your domain, click **Verify**. A new window appears with steps to verify your domain.
```

> 6. Please follow the process from pop-up message or contact administrator of your domain for assistance.

Reviewer: ```suggestion
6. Follow the steps to verify your domain. If you can't complete the steps yourself, ask the administrator of your domain to complete them instead.
```

> 7. After updating the DNS record, click **Verify** to notify Astro to check your DNS record.

Reviewer: ```suggestion
7. Click **Verify** to notify Astro to check your DNS record.
```
